tests/integration/active-model-serializer-test.js

```diff
@@ -368,6 +368,50 @@ test("extractPolymorphic belongsTo", function(assert) {
   });
 });
 
+test("extractPolymorphic belongsTo (weird format)", function(assert) {
+  env.registry.register('adapter:yellowMinion', ActiveModelAdapter);
+  EvilMinion.toString   = function() { return "EvilMinion"; };
+  YellowMinion.toString = function() { return "YellowMinion"; };
+
+  var json_hash = {
+    doomsday_device: {
+      id: 1,
+      name: "DeathRay",
+      evil_minion_id: 12,
+      evil_minion_type: "yellow_minion"
+    },
+    yellow_minions:    [{ id: 12, name: "Alex", doomsday_device_ids: [1] }]
+  };
+  var json;
+
+  run(function() {
+    json = env.amsSerializer.normalizeResponse(env.store, DoomsdayDevice, json_hash, '1', 'findRecord');
+  });
+
+  assert.deepEqual(json, {
+    "data": {
+      "id": "1",
+      "type": "doomsday-device",
+      "attributes": {
+        "name": "DeathRay"
+      },
+      "relationships": {
+        "evilMinion": {
+          "data": { "id": "12", "type": "yellow-minion" }
+        }
+      }
+    },
+    "included": [{
+      "id": "12",
+      "type": "yellow-minion",
+      "attributes": {
+        "name": "Alex"
+      },
+      "relationships": {}
```

Comment:
I basically copy/paste the above test https://github.com/ember-data/active-model-adapter/pull/20/files#diff-a3749e51f929c150f07ac79144078b4fR366, so would that mean there is a bug ?